nixos/modules/services/web-apps/galene.nix

```diff
@@ -0,0 +1,130 @@
+        type = types.str;
+        description = "Galene state directory.";
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "galene";
+        description = "User account under which galene runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "galene";
+        description = "Group under which galene runs.";
+      };
+
+      httpAddress = mkOption {
+        type = types.str;
+        default = "";
+        description = "HTTP listen address for galene.";
+      };
+
+      httpPort = mkOption {
+        type = types.int;
+        default = 8443;
+        description = "HTTP listen port.";
+      };
+
+      staticRootPath = mkOption {
+        type = types.str;
+        default = "${pkgs.galene.static}/static";
+        example = "/var/lib/galene/static";
+        description = "Web server directory.";
+      };
+
+      recordingsRootPath = mkOption {
+        type = types.str;
+        default = defaultRecordingsRootPath;
+        example = "/var/lib/galene/recordings";
+        description = "Recordings directory.";
+      };
+
+      dataRootPath = mkOption {
+        type = types.str;
+        default = defaultDataRootPath;
+        example = "/var/lib/galene/data";
+        description = "Data directory.";
+      };
+
+      groupsRootPath = mkOption {
+        type = types.str;
+        default = defaultGroupsRootPath;
+        example = "/var/lib/galene/groups";
+        description = "Web server directory.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.galene = {
+      description = "galene";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      path = [ pkgs.galene ];
+
+      preStart =
+        ''
+          mkdir -m 0700 -p ${cfg.stateDir}
+          ${optionalString (cfg.recordingsRootPath == defaultRecordingsRootPath) ''
+            mkdir -p ${cfg.recordingsRootPath}
+          ''}
+          ${optionalString (cfg.dataRootPath == defaultDataRootPath) ''
+            mkdir -p ${cfg.dataRootPath}
+          ''}
+          ${optionalString (cfg.groupsRootPath == defaultGroupsRootPath) ''
+            mkdir -p ${cfg.groupsRootPath}
+          ''}
+          chown -R ${cfg.user}:${cfg.group} ${cfg.stateDir}
+        '';
+
+      serviceConfig = {
+        Type = "simple";
+        User = cfg.user;
+        Group = cfg.group;
+        WorkingDirectory = cfg.stateDir;
+        ExecStart = ''${pkgs.galene}/bin/galene \
+        -insecure \
+        -data ${cfg.dataRootPath} \
+        -groups ${cfg.groupsRootPath} \
+        -recordings ${cfg.recordingsRootPath} \
+        -static ${cfg.staticRootPath}'';
+        Restart = "always";
+        # Upstream Requirements
+        LimitNOFILE = 65536;
+      };
+    };
+
+    users.users = mkIf (cfg.user == "galene") {
+      galene = {
+        description = "galene Service";
+        home = cfg.stateDir;
+        createHome = true;
```

Comment:
Is it necessary?

Reply:
Do not use `createHome` to provision directories. It can have a number of problems associated with it.